### PR TITLE
[POAE7-2367] enable some TPC-H queries

### PR DESCRIPTION
**Pull request by harborn**

### CODE REVIEW

File: cider-velox/src/planTransformer/PlanBranches.cpp [hunk 1]
```diff
@@ -33,17 +34,15 @@ void PlanBranches::splitPlanBranches(VeloxPlanNodePtr rootNode) {
...
-  VeloxPlanNodePtr curBranchRoot = branchRootNode;
   VeloxPlanNodeVec curBranch = makePlanBranch(branchRootNode);
   size_t curBranchSize = curBranch.size();
   if (curBranchSize > 0) {
     srcToRootBranchIds_.emplace_back(branchIdx);
     branchMap_[branchIdx] = curBranch;
     VeloxPlanNodePtr branchSourceNode = curBranch[curBranchSize - 1];
     auto branchSrcs = branchSourceNode->sources();
-    if (branchSrcs.empty() or branchSrcs.size() < 2) {
+    if (branchSrcs.empty() || branchSrcs.size() < 2) {
```
Comment (xieqi) [inline cider-velox/src/planTransformer/PlanBranches.cpp:44]: do we need change or to ||, and to &&?

File: cider-velox/src/planTransformer/SequencePlanPattern.h [hunk 2]
```diff
@@ -24,8 +24,10 @@
 #include "PlanPattern.h"
 
 namespace facebook::velox::plugin::plantransformer {
+
 class State;
```
Comment (xieqi) [inline cider-velox/src/planTransformer/SequencePlanPattern.h:28]: need this forward declarations?

**Comment by winningsix** [comment]
This is an ever-green PR?

File: cider-velox/src/planTransformer/PlanTransformer.cpp [hunk 3]
```diff
@@ -351,11 +353,25 @@ PlanTransformerFactory& PlanTransformerFactory::registerPattern(
     std::shared_ptr<PlanPattern> pattern,
     std::shared_ptr<PlanRewriter> rewriter) {
   patternRewriters_.emplace_back(std::make_shared<PatternRewriter>(pattern, rewriter));
+#if 0
```
Comment (xieqi) [inline cider-velox/src/planTransformer/PlanTransformer.cpp:356]: remove debug code

File: cider-velox/src/ArrowConvertorUtils.cpp [hunk 4]
```diff
@@ -90,9 +90,9 @@ int8_t* convertToCider(const ArrowSchema& arrowSchema,
 static void releaseArray(ArrowArray* array) {
   // Free the buffers, for basic types buffer size should be 2 (null and value
   // buffer)
-  free((void*)array->buffers[0]);
-  free((void*)array->buffers[1]);
-  free(array->buffers);
+  // free((void*)array->buffers[0]);
```
Comment (xieqi) [inline cider-velox/src/ArrowConvertorUtils.cpp:93]: why change? Seems no one use this except test.

File: cider-velox/src/CiderVeloxPluginCtx.cpp [hunk 5]
```diff
@@ -25,25 +25,27 @@
 #include "exec/plan/parser/ConverterHelper.h"
 
 namespace facebook::velox::plugin {
+
+static std::mutex s_g_cider_velox_plugin_ctx_mutex;
```
Comment (xieqi) [inline cider-velox/src/CiderVeloxPluginCtx.cpp:29]: a class static memory is better.

**Comment by winningsix** [comment]
Please help update the title to more concrete contexts and fix lint-check failed issue. Thanks!

**Comment by harborn** [comment]
close

**Pull request closed without merging**